### PR TITLE
DEVX-6412: Fixing inaccuracies in WhatsApp Template Message specification

### PR DESCRIPTION
**Pull request by superchilled**
# Description

Updates the WhatsApp Template Message spec for Messages v1 to fix some inaccuracies

# Fixes

* `template.parameters` was incorrectly listed as an array of objects. This corrects it to an array of strings.
* Fixes the example for `template.name` to include a namespace as well as the template name.
* Changes `whatsapp.policy` from being required to optional
* Improves some descriptions

# Checklist

- [x] version number incremented (in the `info` section of the spec)

### CODE REVIEW

File: definitions/messages-olympus.yml [hunk 1]
```diff
@@ -222,18 +222,17 @@ paths:
...
                                 example: deterministic
                                 enum:
                                   - deterministic
-                                description: 'Policy for resolving what language template to use. Please note that WhatsApp deprecated the `fallback` policy in January of 2020, all requests carrying a `fallback` policy will be rejected with a 400 error. As of right now the only valid choice is `deterministic`'
+                                description: 'Policy for resolving what language template to use. As of right now the only valid choice is `deterministic`'
                               locale:
                                 type: string
-                                example: en-US
+                                example: en_US
                                 description: 'The [BCP 47](https://en.wikipedia.org/wiki/IETF_language_tag) language of the template. Vonage will translate the BCP 47 format to the [WhatsApp equivalent](https://developers.facebook.com/docs/whatsapp/message-templates/creation#translations). For examples `en-GB` will be auto-translate to en_GB.'
```
Comment (SMadani) [inline definitions/messages-olympus.yml:236]: Maybe link to [the list of actual supported languages](https://developers.facebook.com/docs/whatsapp/api/messages/message-templates)?

File: definitions/messages-olympus.yml [hunk 2]
```diff
@@ -565,13 +564,14 @@ components:
...
     TemplateParameters:
-      description: List of parameters to populate the template
-      type: object
-      properties:
-        key:
-          type: string
-          description: key/value pair.You can find the full list of supported parameters on WhatsApp's [messages parameters documentation] (https://developers.facebook.com/docs/whatsapp/on-premises/reference/messages#message-templates).
-          example: value
+      type: array
+      description: The parameters are an array of strings, with the first object being used for {{1}} in the template, with the second being {{2}} etc. Only required if the template specified by `name` contains parameters.
```
[the remark this answers is not in the document]
Comment (superchilled) [reply definitions/messages-olympus.yml:568]: Good spot

File: definitions/messages-olympus.yml [hunk 3]
```diff
@@ -222,19 +222,18 @@ paths:
...
-                              - policy
                               - locale
                             properties:
                               policy:
                                 type: string
                                 example: deterministic
                                 enum:
                                   - deterministic
-                                description: 'Policy for resolving what language template to use. Please note that WhatsApp deprecated the `fallback` policy in January of 2020, all requests carrying a `fallback` policy will be rejected with a 400 error. As of right now the only valid choice is `deterministic`'
+                                description: 'Policy for resolving what language template to use. As of right now the only valid choice is `deterministic`'
```
Comment (SMadani) [inline definitions/messages-olympus.yml:232]: Missing full stop at the end.

Comment (SMadani) [inline definitions/messages-olympus.yml:232]:
```suggestion
                                description: 'Policy for resolving what language template to use. As of right now the only valid choice is `deterministic`.'
```

**Review by SMadani — APPROVED** [review]

**Pull request merged**